lib/shopify_api/event_pipe/event_queue.ex

```diff
@@ -61,31 +59,23 @@ defmodule ShopifyAPI.EventPipe.EventQueue do
 
   defp enqueue_event(worker, event, opts) do
     Logger.warn(fn -> "Enqueueing in default queue #{inspect(event)}" end)
-    background_job_impl().(Exq, "default", worker, [event], opts)
+    background_job_enqueue("default", worker, [event], opts)
     {:ok}
   end
 
-  def register(token) do
-    case GenServer.whereis(Exq) do
-      nil ->
-        :timer.sleep(@retry_timer)
-        register(token)
+  def subscribe(token) do
+    background_job_impl().subscribe(token)
+  end
 
-      _res ->
-        Logger.info(fn -> "#{__MODULE__} registering #{AuthToken.create_key(token)}" end)
-        Exq.subscribe(Exq, AuthToken.create_key(token), 1)
-    end
+  defp background_job_enqueue(queue_name, worker, events, opts) do
+    background_job_impl().enqueue(queue_name, worker, events, opts)
   end
 
   defp background_job_impl do
-    pid = GenServer.whereis(Exq)
-
-    if is_pid(pid) do
-      &Exq.enqueue/5
-    else
-      &run_inline/5
-    end
+    Application.get_env(
+      :shopify_api,
+      :background_job_implementation,
+      ShopifyAPI.EventPipe.ExqBackgroundJob
```

Comment:
in the future I imagine this will actually just default to inline... 🤔

Reply:
Interesting. I suppose it gets people up and running with the library sooner.

Reply:
yup, and we are running a fair number of things without it right now :D It has a use, mostly for multi-app large stuff

Reply:
Ah, that is very interesting. Do you think there should be a simple task implementation?